crates/ide/src/goto_definition.rs

```diff
@@ -987,31 +1009,31 @@ fn g() -> <() as Iterator<A = (), B<|> = u8>>::A {}
     }
 
     #[test]
-    fn todo_def_type_for_self() {
+    fn goto_self_param_ty_specified() {
         check(
             r#"
 struct Foo {}
-     //^^^
 
 impl Foo {
-    fn bar(&self<|>) {}
-}
-"#,
-        );
+    fn bar(&self: &Foo) {
```

Comment:
This isn't valid syntax, maybe this does the trick?

```suggestion
    fn bar(self: &Foo) {
```